packages/iTwinUI-react/src/core/Table/Table.tsx

```diff
@@ -403,6 +405,41 @@ export const Table = <
     return flatColumns.some((column) => column.id === SELECTION_CELL_ID);
   }, [columns]);
 
+  const disableUserSelect = React.useCallback(
+    (e: KeyboardEvent) => {
+      if (e.key === 'Shift') {
+        ownerDocument &&
+          (ownerDocument.documentElement.style.userSelect = 'none');
+      }
+    },
+    [ownerDocument],
+  );
+
+  const enableUserSelect = React.useCallback(
+    (e: KeyboardEvent) => {
+      if (e.key === 'Shift') {
+        ownerDocument && (ownerDocument.documentElement.style.userSelect = '');
+      }
+    },
+    [ownerDocument],
+  );
+
+  React.useEffect(() => {
+    // Add onkeydown event listeners (during creation)
+    if (selectionMode === 'multi') {
+      document.addEventListener('keydown', disableUserSelect);
+      document.addEventListener('keyup', enableUserSelect);
+    }
+
+    // Remove onkeydown event listeners (during destruction)
+    return () => {
+      if (selectionMode === 'multi') {
+        document.removeEventListener('keydown', disableUserSelect);
+        document.removeEventListener('keyup', enableUserSelect);
+      }
+    };
+  }, [selectionMode, disableUserSelect, enableUserSelect]);
```

Comment:
Is it okay to merge the on creation logic (429-432) and on destruction logic (435-440) into one useEffect? Or should I separate them into two?

Reply:
It's ok. I think it's a standard practise.